server/src/main/java/org/elasticsearch/cluster/routing/allocation/MoveDecision.java

```diff
@@ -150,16 +150,26 @@ public boolean isDecisionTaken() {
      * returns {@code false} otherwise.  If {@link #isDecisionTaken()} returns {@code false},
      * then invoking this method will throw an {@code IllegalStateException}.
      */
-    public boolean forceMove() {
+    public boolean cannotRemainAndCanMove() {
+        checkDecisionState();
+        return canRemainYes() == false && canMoveDecision == AllocationDecision.YES;
+    }
+
+    /**
+     * Returns {@code true} if the shard cannot remain on its current node and _cannot_ be moved.
+     * returns {@code false} otherwise.  If {@link #isDecisionTaken()} returns {@code false},
+     * then invoking this method will throw an {@code IllegalStateException}.
+     */
+    public boolean cannotRemainAndCannotMove() {
         checkDecisionState();
-        return canRemain() == false && canMoveDecision == AllocationDecision.YES;
+        return canRemainDecision.type() != Type.YES && canMoveDecision != AllocationDecision.YES;
     }
 
     /**
      * Returns {@code true} if the shard can remain on its current node, returns {@code false} otherwise.
      * If {@link #isDecisionTaken()} returns {@code false}, then invoking this method will throw an {@code IllegalStateException}.
      */
-    public boolean canRemain() {
+    public boolean canRemainYes() {
```

Comment:
This change does not seem necessary to me. The `Yes` part is quite obviously implied for a boolean method.